groups/reconcile.go

```diff
@@ -0,0 +1,405 @@
+			log.Printf("skipping updating group settings as group %s has not yet been created\n", groupEmailId)
+			return nil
+		}
+		return fmt.Errorf("unable to retrieve group info for group %s: %v", groupEmailId, err)
+	}
+	if g2.AllowExternalMembers != "true" {
+		if config.DryRun {
+			log.Printf("dry-run : skipping updating group settings\n")
+		} else {
+			_, err := srv.Groups.Patch(groupEmailId, &groupssettings.Groups{
+				AllowExternalMembers: "true",
+			}).Do()
+			if err != nil {
+				return fmt.Errorf("unable to update group info for group %s: %v", groupEmailId, err)
+			} else {
+				log.Printf("> Successfully updated group settings for %s to allow external members\n", groupEmailId)
+			}
+		}
+	}
+	return nil
+}
+
+func addMembersToGroup(service *admin.Service, groupEmailId string, members []string) error {
+	l, err := service.Members.List(groupEmailId).Do()
+	if err != nil {
+		if apierr, ok := err.(*googleapi.Error); ok && apierr.Code == http.StatusNotFound {
+			log.Printf("skipping adding members to group %s as it has not yet been created\n", groupEmailId)
+			return nil
+		}
+		return fmt.Errorf("unable to retrieve members in group %s: %v", groupEmailId, err)
+	}
+
+	for _, m := range members {
+		found := false
+		if len(l.Members) > 0 {
+			for _, m2 := range l.Members {
+				if m2.Email == m {
+					found = true
+					break
+				}
+			}
+		}
+		if found {
+			continue
+		}
+		// We did not find the person in the google group, so we add them
+		if config.DryRun {
+			log.Printf("dry-run : Skipping adding %s to %s\n", m, groupEmailId)
+		} else {
+			_, err := service.Members.Insert(groupEmailId, &admin.Member{
+				Email: m,
+				Role:  "MEMBER",
+			}).Do()
+			if err != nil {
+				return fmt.Errorf("unable to add %s to %s : %v", m, groupEmailId, err)
+			} else {
+				log.Printf("Added %s to %s as a MEMBER\n", m, groupEmailId)
+			}
+		}
+	}
+	return nil
+}
+
+func removeMembersFromGroup(service *admin.Service, groupEmailId string, members []string) error {
+	l, err := service.Members.List(groupEmailId).Do()
+	if err != nil {
+		if apierr, ok := err.(*googleapi.Error); ok && apierr.Code == http.StatusNotFound {
+			log.Printf("skipping removing members group %s as group has not yet been created\n", groupEmailId)
+			return nil
+		}
+		return fmt.Errorf("unable to retrieve members in group %s: %v", groupEmailId, err)
+	}
+
+	for _, m := range l.Members {
+		found := false
+		if len(members) > 0 {
+			for _, m2 := range members {
+				if m2 == m.Email {
+					found = true
+					break
+				}
+			}
+		}
+		if found {
+			continue
+		}
+		// a person was deleted from a group, let's remove them
+		if config.DryRun {
+			log.Printf("dry-run : Skipping removing %s from %s\n", m.Email, groupEmailId)
+		} else {
+			err := service.Members.Delete(groupEmailId, m.Email).Do()
+			if err != nil {
+				return fmt.Errorf("unable to remove %s from %s : %v", m.Email, groupEmailId, err)
+			} else {
```

Comment:
else is not needed:

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow